src/api/config.js

```diff
@@ -1,31 +1,64 @@
 'use strict'
 
-const argCommand = require('../cmd-helpers').argCommand
+const streamifier = require('streamifier')
 
 module.exports = (send) => {
   return {
-    get: argCommand(send, 'config'),
-    set (key, value, opts, cb) {
-      if (typeof (opts) === 'function') {
-        cb = opts
+    get (key, callback) {
+      if (typeof key === 'function') {
+        callback = key
+        key = undefined
+      }
+
+      if (!key) {
+        return send('config/show', null, null, null, true, callback)
+      }
+
+      return send('config', key, null, null, (err, result) => {
+        if (err) {
+          return callback(err)
+        }
+        callback(null, result.Value)
+      })
+    },
+    set (key, value, opts, callback) {
+      if (typeof opts === 'function') {
+        callback = opts
         opts = {}
       }
+      if (typeof key !== 'string') {
+        return callback(new Error('Invalid key type'))
+      }
+
+      if (typeof value !== 'object' &&
+          typeof value !== 'boolean' &&
+          typeof value !== 'string') {
+        return callback(new Error('Invalid value type'))
+      }
 
-      if (typeof (value) === 'object') {
+      if (typeof value === 'object') {
         value = JSON.stringify(value)
         opts = { json: true }
-      } else if (typeof (value) === 'boolean') {
+      }
+
+      if (typeof value === 'boolean') {
         value = value.toString()
         opts = { bool: true }
       }
 
-      return send('config', [key, value], opts, null, cb)
-    },
-    show (cb) {
-      return send('config/show', null, null, null, true, cb)
+      return send('config', [key, value], opts, null, callback)
     },
-    replace (file, cb) {
-      return send('config/replace', null, null, file, cb)
+    replace (config, callback) {
+      // Its a path
+      if (typeof config === 'string') {
+        return send('config/replace', null, null, config, callback)
+      }
+
+      // Its a config obj
+      if (typeof config === 'object') {
+        config = streamifier.createReadStream(new Buffer(JSON.stringify(config)))
+        return send('config/replace', null, null, config, callback)
+      }
```

Comment:
Just missing to make the config.replace tests to work. It seems that this was never working (see https://github.com/ipfs/js-ipfs-api/blob/master/test/api/config.spec.js#L123-L133), unfortunately.

The current problem is that the config never gets replaced and after we send the replace, the content type of the following config.get requests becomes text/plain instead of application/json